pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -895,16 +897,16 @@ func (cr *PerconaXtraDBCluster) ProxySQLEnabled() bool {
 	return cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled
 }
 
-func (s *PerconaXtraDBClusterStatus) ClusterStatus(inProgress bool) AppState {
+func (s *PerconaXtraDBClusterStatus) ClusterStatus(deleted bool) AppState {
 	switch {
-	case s.PXC.Status == AppStateError || s.ProxySQL.Status == AppStateError || s.HAProxy.Status == AppStateError:
-		return AppStateError
-	case inProgress || s.PXC.Status == AppStateInit || s.ProxySQL.Status == AppStateInit || s.HAProxy.Status == AppStateInit:
-		return AppStateInit
-	case s.PXC.Status == AppStateReady:
+	case deleted || s.PXC.Status == AppStateStopping || s.ProxySQL.Status == AppStateStopping || s.HAProxy.Status == AppStateStopping:
+		return AppStateStopping
+	case s.PXC.Status == AppStatePaused && ((s.ProxySQL.Status == AppStatePaused || s.HAProxy.Status == AppStatePaused) || true):
+		return AppStatePaused
+	case s.PXC.Status == AppStateReady && ((s.ProxySQL.Status == AppStateReady || s.HAProxy.Status == AppStateReady) || true):
```

Comment:
`((s.ProxySQL.Status == AppStatePaused || s.HAProxy.Status == AppStatePaused) || true)` condition is always true, it impossible to get false from it

Reply:
fixed in https://github.com/percona/percona-xtradb-cluster-operator/pull/894/commits/650a935ad7d1abf5850c129ac42584562e06c069